tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -200,6 +201,59 @@ namespace Tensile
                 }
             };
 
+            struct SynchronizerSizeCheck
+                : public Predicate_CRTP<SynchronizerSizeCheck, ContractionProblemGemm>
+            {
+                enum
+                {
+                    HasIndex = true,
+                    HasValue = true
+                };
+                size_t             index;
+                std::array<int, 5> value;
+
+                SynchronizerSizeCheck() = default;
+                SynchronizerSizeCheck(size_t index, std::array<int, 5> value)
+                    : index(index)
+                    , value(value)
+                {
+                }
+
+                static std::string Type()
+                {
+                    return "SynchronizerSizeCheck";
+                }
+
+                virtual bool operator()(ContractionProblemGemm const& problem) const override
+                {
+                    // WorkGroup numbers x global write batch x Wave numbers
+                    // M/MT0 x N/MT1 x NumElementsPerThread/StoreVectorWidth x x Wavenumbers
+                    bool ret = (std::ceil(static_cast<float>(problem.freeSizeA(0)) / value[0])
+                                * std::ceil(static_cast<float>(problem.freeSizeB(0)) / value[1]))
+                                   * (value[2] / value[3]) * (value[4] / 64)
+                               <= 1024;
+                    if(problem.groupedGemm())
+                        ret = ret && (problem.groupedGemmCount() <= 16);
+
+                    return ret;
+                }
+
+                virtual bool debugEval(ContractionProblemGemm const& problem,
+                                       std::ostream&                 stream) const override
+                {
+                    return debugEvalCmp(
+                        problem,
+                        stream,
+                        "prob",
+                        (std::ceil(static_cast<float>(problem.freeSizeA(0)) / value[0])
+                         * std::ceil(static_cast<float>(problem.freeSizeB(0)) / value[1]))
+                            * (value[2] / value[3]),
```

Comment:
you should sync formulat with operator()

Reply:
sure